src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1131,6 +1131,15 @@ public static bool IsParameterTypeContext(this SyntaxTree syntaxTree, int positi
             return false;
         }
 
+        public static bool IsPossibleTupleElementNameContext(
+            this SyntaxTree syntaxTree, int position, SyntaxToken tokenOnLeftOfPosition, CancellationToken cancellationToken)
+        {
+            var token = tokenOnLeftOfPosition;
+            token = token.GetPreviousTokenIfTouchingWord(position);
+
+            return token.IsKind(SyntaxKind.OpenParenToken, SyntaxKind.CommaToken);
```

Comment:
This is too aggressive.  We should be checking that the open paren is specifically part of a tuple literal, or a parenthesized expression.  I'm not sure about the comma.  But it's likely that by that time, we can just see that hte comma is parented by a tuple literal.

Here's the case where the above code is bad (and we should have a command handler test for this).  Say the user writes:

```Foo(X<colon>``` or ```Foo(args, X<colon>```.

Based on your code, we would consider both of these cases to be a TupleElementName context.  but in both cases we want to complete out the colon.